javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplate.java

```diff
@@ -544,7 +544,11 @@ public final void write(ClientMethod clientMethod, JavaType typeBlock) {
                 break;
 
             case LongRunningBeginSync:
-                generateLongRunningBeginSync(clientMethod, typeBlock, restAPIMethod, settings);
+                if (settings.isDataPlaneClient()) {
+                    generateProtocolLongRunningBeginSync(clientMethod, typeBlock);
+                } else {
+                    generateLongRunningBeginSync(clientMethod, typeBlock, restAPIMethod, settings);
+                }
```

Comment:
may need a `isSyncStackEnabled` check here or below?

Reply:
Yep, added the check.